### DISCUSSION TITLE
Invalid Certificate Signing Request

### DISCUSSION THREAD

**Discussion by reyhanwidjaja** [micromdm/micromdm#832] [Q&A] [answered]
<img width="987" alt="Screen Shot 2022-06-27 at 16 03 57" src="https://user-images.githubusercontent.com/72331112/175902598-1fd70f1b-a9e8-439b-8e9b-1054cb6d1f7b.png">

I'm having this error when try upload the generated csr file, based on this documentation https://github.com/micromdm/micromdm/blob/main/docs/user-guide/quickstart.md#mdmcertdownload, now I'm stuck at step number 6. Does anyone have any solution for this?

**Comment by jessepeterson** [comment, marked as answer]
This is probably related to #833 which is an Apple-side regression of an issue that affected APNs cert upload requests. You can try renaming the file to `.rtf`.

**Reply by Tieantono** [reply]
I did this, along with renaming it to `.txt` too, but it still won't accept the CSR file. Is `mdmcert.download.push.csr` the one that we should upload to identity.apple.com or am I supposedly to upload another file from decryption result?

Also in my case, I was using `admin@mycompany.com` when submitting a CSR to mdmcert.download's API, but I signed in with my Apple ID using `tieantono@mycompany.com` on identity.apple.com, does it matter?

**Reply by jessepeterson** [reply]
Just curious what version of MicroMDM you're using? v1.8 included the fix for #723 which presented itself similar to what you're seeing here. I wonder if that's what you're issue is?

**Reply by reyhanwidjaja** [reply]
It turns out i uploaded the wrong one, what I uploaded was mdmcert.download.push.csr. The solution for this problem is to upload mdmcert.download.push.req (change ext to .rtf) to https://identity.apple.com

**Reply by jessepeterson** [reply]
@Tieantono mdmcert.download has no connection to Apple IDs — it's completely separate. Which Apple ID you use to upload the certificate to identity.apple.com is unrelated to the email address mdmcert.download uses to issue certificates to.
